psqlgraph/mocks.py

```diff
@@ -0,0 +1,446 @@
+        If override is provided, use override value as property-value
+
+        :param name: property name
+        :type name: String
+        :param override: property value
+        :type override: type
+        :return: Tuple(str, type)
+        """
+        if override is not None:
+            return name, override
+
+        return name, self.type_factories[name].random_value()
+
+    def resolve_type(self, definition):
+        if 'enum' in definition:
+            return EnumRand(definition['enum'])
+
+        if 'type' in definition:
+            return TypeRandFactory.get_randomizer(definition)
+
+        if 'oneOf' in definition:
+            return self.resolve_type(definition['oneOf'][0])
+
+        if 'anyOf' in definition:
+            return self.resolve_type(definition['anyOf'][0])
+
+        return StringRand(definition)
+
+
+class NodeFactory(object):
+    def __init__(self, models, schema):
+        self.models = models
+        self.schema = schema
+        self.property_factories = {
+            label: PropertyFactory(node_def['properties'])
+            for label, node_def in schema.items()
+        }
+
+    def create(self, label, override=None, all_props=False):
+        """
+        Create a minimal node of `label` type with only required properties
+        being set. Override the values of properties by providing an `override`
+        dictionary. If all properties should be set to some value, set
+        `all_props` to True
+
+        :param label: target node type
+        :type label: String
+        :param override: properties to override
+        :type override: Dict
+        :param all_props: set all properties to an arbitrary value or not
+        :type all_props: Boolean
+
+        :return: psqlgraph.Node object
+        """
+        if not override:
+            override = {}
+
+        node_json = {
+            'node_id': override.pop('node_id', str(uuid.uuid4())),
+            'acl': override.pop('acl', []),
+            'properties': {},
+            'system_annotations': override.pop('system_annotations', {}),
+        }
+
+        if all_props:
+            prop_list = self.schema[label].get('properties', [])
+        else:
+            prop_list = self.schema[label].get('required', [])
+
+        for prop in prop_list:
+            if prop == 'type':
+                continue
+
+            try:
+                _, value = self.property_factories[label].create(
+                    prop, override.get('properties', {}).get(prop))
+            except KeyError:
+                # print("No factory for property: '{}'".format(prop))
+                continue
+
+            node_json['properties'][prop] = value
+
+        override.pop('properties', None)
+        node_json.update(override)
+        node_cls = self.models.Node.get_subclass(label)
+
+        return node_cls(**node_json)
+
+
+class GraphFactory(object):
+    def __init__(self, models, dictionary):
```

Comment:
can we allow passing a `project_id` as part of the constructor which will at least be used to set the `project_id` field on the nodes

Reply:
👍 good feature. done